endogenous/scripts/end_transport.py

```diff
@@ -111,36 +113,92 @@ def transport(string):
                 capital_cost = capital_cost_solar,
-                marginal_cost = 0.01,
+                marginal_cost = solar.at['VOM','value'],
                 p_max_pu = CF_solar)
     
-    
+    network.add("Bus", "battery bus")
+
+    df_costs = pd.read_csv(string + 'resources/costs_' + str(Nyear) + '.csv', sep=',', index_col=0)
+    #set first column as second index and keep original index
+    df_costs.set_index(df_costs.columns[0],append=True,inplace=True)
+    #set parameter/value names as column name, min_count=1 fills not existing values with NaN
+    costs = df_costs.loc[:,"value"].unstack(level=1).groupby("technology").sum(min_count=1)
+    network.add("Store", "battery storage",
+                bus="battery bus",
+                e_nom_extendable=True,
+                e_cyclic = True,
+                capital_cost=costs.at['battery storage', 'investment'],
+                lifetime = costs.at['battery storage', 'lifetime'])
+    network.add("Carrier",'battery')
+    network.add("Link",
+                "battery charger",
+                bus0 = "electricity bus",
+                bus1 = 'battery bus',
+                carrier = 'battery charger',
+                p_nom_extendable = 'True',
+                efficiency = costs.at['battery inverter', 'efficiency']**0.5,
+                capital_cost = costs.at['battery inverter', 'investment']*(annuity(costs.at['battery inverter', 'lifetime'], 0.07)+costs.at['battery inverter', 'FOM']/100),
+                lifetime = costs.at['battery inverter', 'lifetime'])
+    network.add("Link",
+                "battery discharger",
+                bus0 = "battery bus",
+                bus1 = 'electricity bus',
+                carrier = 'battery discharger',
+                p_nom_extendable = 'True',
+                efficiency = costs.at['battery inverter', 'efficiency']**0.5,
+                capital_cost = costs.at['battery inverter', 'investment']*(annuity(costs.at['battery inverter', 'lifetime'], 0.07)+costs.at['battery inverter', 'FOM']/100),
+                lifetime = costs.at['battery inverter', 'lifetime'])
     network.add("Carrier", "vehicle")
     
     network.add("Bus", 
                 "vehicle bus",
                 carrier="vehicle")
     
+    network.add("Carrier", "co2", co2_emissions=1.)
+
+    network.add("Bus", 
+                "co2 atmosphere", 
+                carrier="co2")
+    
+    # convert Mt to tCO2
+    co2_totals = 1e6 * pd.read_csv(snakemake.input.co2_totals_name, index_col=0)
+
+    co2_limit = co2_totals.loc[country_short, 'road non-elec']
+    co2_limit = co2_limit * limit
+    #co2_limit = co2_limit * Nyear
+    target = 0#3.392052e+03/8760
     
+    network.add("Store",
+                "co2 atmosphere",
+                e_initial= 0, 
+                e_nom = co2_limit,
+                e_min_pu = -1,
+                bus="co2 atmosphere", 
+                carrier="co2")
```

Comment:
Ok, I believe you mentioned this to simplify the code implemented some weeks ago for the local/nodal co2 atmosphere found here: https://github.com/ricnogfer/pypsa-eur/blob/local_co2_atmosphere/scripts/prepare_sector_network.py#L598-L606. Will refactor the code accordingly then.

Reply:
Yes, I do. 
'e_max_pu' is the maximum filling level per unit (that is, the maximum value is 1)
'e_nom' is the energy capacity of the store, in your case the maximum CO2 that can be stored
I think your former implementation does not work because it implies maximum Co2 stored is 1.